test/RibbonThetaYearnVault.ts

```diff
@@ -0,0 +1,2964 @@
+    userSigner: SignerWithAddress,
+    ownerSigner: SignerWithAddress,
+    feeRecipientSigner: SignerWithAddress;
+
+  // Parameters
+  let tokenName = params.tokenName;
+  let tokenSymbol = params.tokenSymbol;
+  let tokenDecimals = params.tokenDecimals;
+  let minimumSupply = params.minimumSupply;
+  let asset = params.asset;
+  let depositAsset = params.depositAsset;
+  let collateralAsset = params.collateralAsset;
+  let depositAmount = params.depositAmount;
+  let premiumDiscount = params.premiumDiscount;
+  let managementFee = params.managementFee;
+  let performanceFee = params.performanceFee;
+  // let expectedMintAmount = params.expectedMintAmount;
+  let auctionDuration = params.auctionDuration;
+  let isPut = params.isPut;
+  let initialSharePrice = params.initialSharePrice;
+
+  // Contracts
+  let strikeSelection: Contract;
+  let volOracle: Contract;
+  let optionsPremiumPricer: Contract;
+  let gnosisAuction: Contract;
+  let vaultLifecycleLib: Contract;
+  let vault: Contract;
+  let oTokenFactory: Contract;
+  let defaultOtoken: Contract;
+  let assetContract: Contract;
+  let collateralContract: Contract;
+  let decimalDiff: BigNumber;
+  let collateralPricerSigner: Contract;
+
+  // Variables
+  let defaultOtokenAddress: string;
+  let firstOptionStrike: BigNumber;
+  let firstOptionPremium: BigNumber;
+  let firstOptionExpiry: number;
+  let secondOptionStrike: BigNumber;
+  let secondOptionExpiry: number;
+
+  describe(`${params.name}`, () => {
+    let initSnapshotId: string;
+    let firstOption: Option;
+    let secondOption: Option;
+
+    const rollToNextOption = async () => {
+      await vault.connect(ownerSigner).commitAndClose();
+      await time.increaseTo((await getNextOptionReadyAt()) + 1);
+      await strikeSelection.setDelta(params.deltaFirstOption);
+      await vault.connect(ownerSigner).rollToNextOption();
+    };
+
+    const rollToSecondOption = async (settlementPrice: BigNumber) => {
+      const oracle = await setupOracle(params.underlyingPricer, ownerSigner);
+
+      await setOpynOracleExpiryPriceYearn(
+        params.asset,
+        oracle,
+        settlementPrice,
+        collateralPricerSigner,
+        await getCurrentOptionExpiry()
+      );
+      await strikeSelection.setDelta(params.deltaSecondOption);
+      await vault.connect(ownerSigner).commitAndClose();
+      await time.increaseTo((await vault.nextOptionReadyAt()).toNumber() + 1);
+      await vault.connect(ownerSigner).rollToNextOption();
+    };
+
+    const getNextOptionReadyAt = async () => {
+      const optionState = await vault.optionState();
+      return optionState.nextOptionReadyAt;
+    };
+
+    const getCurrentOptionExpiry = async () => {
+      const currentOption = await vault.currentOption();
+      const otoken = await getContractAt("IOtoken", currentOption);
+      return otoken.expiryTimestamp();
+    };
+
+    before(async function () {
+      // Reset block
+      await network.provider.request({
```

Comment:
Is there some way to avoid doing `hardhat_reset`? Calling reset breaks the test coverage tool.

Alternatively, we can use [this solution to fork the state](https://github.com/sc-forks/solidity-coverage/issues/574#issuecomment-856330142).